test/features/build_server/compiler_interface_test.dart

```diff
@@ -0,0 +1,319 @@
+      ...List.filled(
+        compiledSourceFiles.length,
+        CompileStatus.compiling,
+      ),
+      CompileStatus.compiling,
+      CompileStatus.linking,
+      CompileStatus.waiting,
+      CompileStatus.done
+    ];
+
+    expect(
+      messages.map((e) => e.status),
+      orderedEquals(expectedMessageStates),
+    );
+  });
+
+  test('Emits a state error when not all files are updated', () async {
+    await expectLater(
+      _buildProject(
+        interface!,
+        files,
+        fileMap,
+        config1,
+        cache!,
+        skipUpdateFiles: true,
+      ),
+      throwsStateError,
+    );
+  });
+
+  group('Worker', () {
+    setUp(() {
+      WorkerInterface.instance = null;
+    });
+
+    test('can be started exactly once', () {
+      final receivePort = ReceivePort();
+      final sendPort = receivePort.sendPort;
+
+      expect(() => WorkerInterface.start(sendPort, receivePort), returnsNormally);
+      expect(() => WorkerInterface.start(sendPort, receivePort), throwsStateError);
+    });
+
+    test('forwards delegation calls', () async {
+      final testReceive = ReceivePort();
+      final receivePort = ReceivePort();
+      final sendPort = testReceive.sendPort;
+
+      WorkerInterface.start(sendPort, receivePort);
+
+      final worker = WorkerInterface.instance!;
+      final compiling = VHook(0);
+      final linking = VHook(0);
+      final waiting = VHook(0);
+      final done = VHook<bool?>(null);
+
+      testReceive.listen((message) {
+        if (message is InterfaceMessage && message.name == 'stateUpdate') {
+          final msg = message.args as CompileStatusMessage;
+          switch (msg.status) {
+            case CompileStatus.compiling:
+              compiling.set(compiling.value + 1);
+              break;
+            case CompileStatus.linking:
+              linking.set(linking.value + 1);
+              break;
+            case CompileStatus.waiting:
+              waiting.set(waiting.value + 1);
+              break;
+            case CompileStatus.done:
+              done.set(true);
+              break;
+            case CompileStatus.failed:
+              break;
+          }
+        }
+      });
+
+      worker.didStartCompilation();
+      worker.isCompiling(0);
+      worker.isCompiling(0.5);
+      worker.isCompiling(1);
+      worker.didFinishCompilation();
+      worker.didStartLinking();
+      worker.didFinishLinking();
+      worker.done();
+
+      await done.awaitValue(Duration(milliseconds: 100));
+
+      done.expect(isTrue);
+      compiling.expect(equals(5));
+      linking.expect(equals(1));
+      waiting.expect(equals(1));
+    });
+
+    test('can be initiated once', () async {
+      final testReceive = ReceivePort();
+      final receivePort = ReceivePort();
+      final sendPort = testReceive.sendPort;
+      final testSend = receivePort.sendPort;
+
+      WorkerInterface.start(sendPort, receivePort);
+
+      int responses = 0;
```

Comment:
Wee need an `expectValue`

Reply:
[https://github.com/Coronon/flyde/pull/9#issuecomment-914277597](https://github.com/Coronon/flyde/pull/9#issuecomment-914277597)

Reply:
I do not agree on this. Adding an unnecessary layer of complexity on top of something simple doesn't sound like a good style choice to me. When reading the code in the future someone would ask himself why the `VHook` has been used which makes the test case harder to maintain.

Reply:
This does look out of place though. All other tests in this fashion utilize `VHook`, so why should we not use it here. If you really don't see the point, I am more than willing to add an `expectAfter` method.

Reply:
I will add `VHook` because you let me decide the other conversation. But still, in my opinion, adding a wrapper for no functionality is not a good style.